utils/land-patterns.stanza

```diff
@@ -881,31 +881,53 @@ public pcb-landpattern dfn-landpattern ( n-pad:Int, pad-pitch:Double, pad-dims:[
   make-dfn-landpattern(n-pad,pad-pitch,pad-dims,pad-x0,body-dims)
   pad p[n-pad + 1] : smd-pad(hs[0],hs[1]) at loc(0.0, 0.0)  
 
-; returns the letter corresponding to a row in a bga package, for example:
-; A, B, C, D, E, E, F, G, H, J, K, L, M, N, P, R, T, U, V, W, Y
-; AA, AB, ... etc
-public defn red-alph-letter (idx:Int):
-  val table = "ABCDEFGHJKLMNPRTUVWY"
+; Returns the name of a row given its index and a list of omitted
+; characters.
+; 
+; The algorithm is a realization of this description from JEDEC JC-11:
+;
+; > When a package is viewed ... with the A1 corner in the upper left position, 
+; > terminal rows are lettered from top to bottom starting with A and 
+; > terminal columns are numbered from left to right starting with 1. 
+; > ... For packages with greater than 20 rows (i.e.,  following rows W and Y),  
+; > the 21st row is designated AA and subsequent rows are designated AB, AC, etc.
+;
+; The modification is we abstract over the omitted characters to support non-
+; standard naming conventions. 
+;
+public defn bga-row-name (idx:Int, omitted:String) -> Ref:
```

Comment:
This algorithm seems to be duplicated later in the same file in the `bga-pad-names` function. Please consolidate the two code blocks into one.